src/argon2_fmt_plug.c

```diff
@@ -253,14 +303,30 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 #ifdef _OPENMP
 #pragma omp parallel for
 #endif
-	for (i = 0; i < count; i++)
-		argon2_hash(saved_salt.t_cost, saved_salt.m_cost, saved_salt.lanes,
-		            saved_key[i], saved_len[i],
-		            saved_salt.salt, saved_salt.salt_length,
-		            crypted[i], saved_salt.hash_size,
-		            NULL, 0,
-		            saved_salt.type, ARGON2_VERSION_NUMBER);
-
+	for (i = 0; i < count; i++) {
+		argon2_context context;
+
+		context.out = (uint8_t*)crypted[i];
+		context.outlen = saved_salt.hash_size;
+		context.pwd = (uint8_t*)saved_key[i];
+		context.pwdlen = saved_len[i];
+		context.salt = (uint8_t*)saved_salt.salt;
+		context.saltlen = saved_salt.salt_length;
+		context.secret = NULL;
+		context.secretlen = 0;
+		context.ad = NULL;
+		context.adlen = 0;
+		context.t_cost = saved_salt.t_cost;
+		context.m_cost = saved_salt.m_cost;
+		context.lanes = saved_salt.lanes;
+		context.threads = saved_salt.lanes;
+		context.allocate_cbk = &allocate;
+		context.free_cbk = &deallocate;
+		context.flags = ARGON2_DEFAULT_FLAGS;
+		context.version = ARGON2_VERSION_NUMBER;
```

Comment:
Maybe use struct initialized with named fields here, which is a C feature that we already use in the tree anyway (and for a related purpose). Like this:

```c
argon2_context context = {
        .out = (uint8_t*)crypted[i],
and so on
};
```

omitting the fields where we initialize to NULL or 0. That way, we'll be sure to initialize the entire struct even if more fields are added later.

Our existing usage is e.g.:

```c
yescrypt_params_t params = {.N = cur_salt->N, .r = cur_salt->r, .p = cur_salt->p};
```